### PR TITLE
Refactor code inspector

### PR DESCRIPTION
**Pull request by larshp**
Changes:
* method GET_INSPECTION deleted, not called from anywhere
* delete class ZCL_ABAPGIT_ABAP_UNIT_TESTS, only used from CI, refactored in https://github.com/abapGit/CI/pull/57
* delete class ZCL_ABAPGIT_SYNTAX_CHECK, the code inspector class is used instead
* delete class ZCL_ABAPGIT_ADHOC_CODE_INSP, not needed anymore, if needed a new method RUN_CHECK can be added to code inspector class instead of a new class
* move `variant` to importing parameter in RUN instead of in the constructor

Sidetracked from #2127, with these refactorings it is possible to move `MV_CODE_INSPECTOR_SUCCESSFUL` from ZCL_ABAPBIT_REPO to ZCL_ABAPGIT_CODE_INSPECTOR, as there is now one instance per package. Suggest adding a parameter SAVE to RUN, if it should save the status in an attribute, then its possible to call `factory->get_code_inspector( package )->is_successful( )` from PUSH in repo class

### CODE REVIEW

**Comment by larshp** [comment]
@christianguenter2 do you want to review this PR?

**Comment by christianguenter2** [comment]
Sure. I'll take a look asap.

File: src/ui/zcl_abapgit_gui_page_syntax.clas.abap [hunk 1]
```diff
@@ -32,9 +32,9 @@ CLASS ZCL_ABAPGIT_GUI_PAGE_SYNTAX IMPLEMENTATION.
 
     DATA: li_syntax_check TYPE REF TO zif_abapgit_code_inspector.
 
-    li_syntax_check = zcl_abapgit_factory=>get_syntax_check( mo_repo->get_package( ) ).
+    li_syntax_check = zcl_abapgit_factory=>get_code_inspector( mo_repo->get_package( ) ).
 
-    mt_result = li_syntax_check->run( ).
+    mt_result = li_syntax_check->run( 'SYNTAX_CHECK' ).
```
Comment (christianguenter2) [inline src/ui/zcl_abapgit_gui_page_syntax.clas.abap]: What do you think about ectracting this to a constants structure? In that way we can easily keep track of the variants we use.

Comment (larshp) [reply src/ui/zcl_abapgit_gui_page_syntax.clas.abap]: *shrug*

**Pull request merged**